### ISSUE TITLE
Authentication error / initial bootstrap

### ISSUE THREAD

**Issue by jo** [eHealthAfrica/kazana#44]
When starting with an empty database I get an error message like this:

```
Err! log 2015-09-08T12:39:21.006Z [ 'myapp', 'transform', 'bookkeeping' ] { [unauthorized: Name or password is incorrect.]
Err! log   status: 401,
Err! log   name: 'unauthorized',
Err! log   message: 'Name or password is incorrect.',
Err! log   error: true,
Err! log   reason: 'Name or password is incorrect.' }
```

Steps to reproduce:

1. Check out [kazana-example](https://github.com/eHealthAfrica/kazana-example)
2. Delete `.db` directory
3. Start the app

**Comment by gr2m** [comment]
looking into this now

**Comment by gr2m** [comment]
args :( Looks like a PouchDB-Server issue. That will need some investigation. How much of a blocker is that?

This is what I get when connected to CouchDB:

```
$ kazana_backend__name="couchdb" kazana_backend__location=http://localhost:5984 kazana_log__level=verbose npm start

> kazana-example@ start /Users/gregor/Projects/eHealth/kazana/example
> kazana

Connected to CouchDB at http://localhost:5984
view?
Info log 2015-09-08T14:36:40.902Z [ 'myapp', 'indices' ] Indices created in: integrated-data
Info log 2015-09-08T14:36:40.920Z [ 'myapp' ] Server running at http://gregor.local:5000
Info log 2015-09-08T14:36:40.934Z [ 'myapp', 'bootstrap' ] successfully connected to http://localhost:5984
Info log 2015-09-08T14:36:40.941Z [ 'kazana', 'bootstrap' ] successfully connected to http://localhost:5984
Err! log 2015-09-08T14:36:41.018Z [ 'myapp', 'transform', 'bookkeeping' ] { [not_found: missing]
Err! log   status: 404,
Err! log   name: 'not_found',
Err! log   message: 'missing',
Err! log   error: true,
Err! log   reason: 'missing' }
Info log 2015-09-08T14:36:41.196Z [ 'myapp', 'bootstrap' ] users created or updated: kazana-admin@example.com, kazana-analyst@example.com, kazana-data-provider@example.com
Info log 2015-09-08T14:36:41.197Z [ 'myapp', 'bootstrap' ] done
Info log 2015-09-08T14:36:41.231Z [ 'kazana', 'bootstrap' ] _security set on database my-integrated-data.
Info log 2015-09-08T14:36:41.233Z [ 'kazana', 'bootstrap' ] _security set on database my-raw-data.
Info log 2015-09-08T14:36:41.234Z [ 'kazana', 'bootstrap' ] _design docs added or updated on database my-raw-data: 1.
Info log 2015-09-08T14:36:41.234Z [ 'kazana', 'bootstrap' ] done
```

**Comment by jo** [comment]
not a blocker

**Comment by gr2m** [comment]
Looks like a racing condition, I think it tries to bootstrap pending raw data before the admin user has been even created:

```
[info] GET /my-raw-data/ 401 - 127.0.0.1
[info] GET /my-integrated-data/ 401 - 127.0.0.1
[info] GET /my-integrated-data/ 401 - 127.0.0.1
[info] GET / 200 - 127.0.0.1
[info] GET /_session 200 - 127.0.0.1
[info] PUT /_config/admins/kazana 200 - 127.0.0.1
```

This might also explain the difference between CouchDB/PouchDB Server in my case. I'll on it

**Comment by gr2m** [comment]
partly fixed by ec30eeb
